src/vmm/src/vstate/vm/mod.rs

```diff
@@ -39,43 +38,25 @@ pub enum VmError {
     Arch(#[from] ArchVmError),
 }
 
-/// A wrapper around creating and using a VM.
+/// Structure containing all relevant things for a virtual machine
 #[derive(Debug)]
 pub struct Vm {
     fd: VmFd,
-
-    // Arm specific fields.
-    // On aarch64 we need to keep around the fd obtained by creating the VGIC device.
-    #[cfg(target_arch = "aarch64")]
-    irqchip_handle: Option<GICDevice>,
+    arch: ArchVm,
 }
 
 /// Contains Vm functions that are usable across CPU architectures
 impl Vm {
     /// Create a new `Vm` struct.
     pub fn new(kvm: &Kvm) -> Result<Self, VmError> {
-        // Create fd for interacting with kvm-vm specific functions.
-        let vm_fd = kvm.fd.create_vm().map_err(VmError::VmFd)?;
-
-        #[cfg(target_arch = "aarch64")]
-        {
-            Ok(Vm {
-                fd: vm_fd,
-                irqchip_handle: None,
-            })
-        }
+        let fd = kvm.fd.create_vm().map_err(VmError::VmFd)?;
```

Comment:
As discussed offline, let's remove this `Vm` type and move this one line of fd creation into each `ArchVm`.

Reply:
Yeah, with just the `fd` shared between architectures there's not much point in having this. Originally I had also moved the `GuestMemoryMmap` stuff into `Vm`, but since that never made the PR there's no point in keeping around `struct Vm` in preparation (of something which might never happen - and if it does we can always change it again)